### ISSUE TITLE
SoftAP read function issue.

### ISSUE THREAD

**Issue by Suprazz** [particle-iot/device-os#1131]
Two continuous `body->read` calls in page handler of the SoftAP application won't return consecutive data except if the read is 1152 bytes long.

Example: This work perfectly:
```c
uint8_t contentData[1152];
uint8_t contentData2[1152];
int datalength = body->read(contentData, sizeof(contentData));
int datalength2 = body->read(contentData2, sizeof(contentData));
```

With this code, there is data lost between the 2 buffers.
```c
uint8_t contentData[512];
uint8_t contentData2[512];
int datalength = body->read(contentData, sizeof(contentData));
int datalength2 = body->read(contentData2, sizeof(contentData));
```

### Refer to this thread for more info: 
https://community.particle.io/t/firmware-upgrade-via-softap/26198/11

---

Completeness:
- [x] Minimum test case added
- [x] Device, system and user firmware versions stated (as of 0.6.0)
- [ ] update documentation
- [ ] on hold for future release of WICED

**Comment by m-mcgowan** [comment]
Thank you for the  issue report. We'll investigate and try to determine the cause of the issue, and in the meantime we will update the documentation to reflect this use case.

**Comment by bkobkobko** [comment]
The Community thread is here:

http://community.particle.io/t/firmware-upgrade-via-softap/26198

The work-around is obivously to use a 1152-byte (TCP_MSS) sized buffer.

The inline read function looks right to me so I think it is something in the WICED source that you can't see without an NDA.

**Comment by m-mcgowan** [comment]
Thanks bko. The WICED http server is implemented by using TCP as packets rather than as a stream (which is performant, but the wrong abstraction imho.) So it doesn't surprise me these kind of things crop up as a result. 

I'm not sure we can fix it without rewriting the http server, but we can at least document it and hope WICED have a fix in a future release. We are expecting a new drop from them soon.

**Comment by avtolstoy** [comment]
This should be resolved as of 0.7.0+.
